tests/_enable_polkit_debugging.pm

```diff
@@ -0,0 +1,23 @@
+# vi: set shiftwidth=4 tabstop=4 expandtab:
+use base 'installedtest';
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    # Install an additional polkit rule which prints debug information for
+    # every authorisation request. This should help debug any problems which are
+    # found in tests, especially unexpected authorisation dialogs (where do they
+    # come from?).
+    $self->root_console();
+    assert_script_run('echo "polkit.addRule(function(action, subject) { polkit.log(\"action=\" + action); polkit.log(\"subject=\" + subject); return polkit.Result.NOT_HANDLED; });" > /etc/polkit-1/rules.d/10-debugging.rules', 30);
```

Comment:
I'm not sure how keen on esoteric Perl constructs you are, but did you know that you can use pretty much any pair of delimiters for quoted strings? If you wanted to avoid having to escape the nested double-quotes you could write:

```perl
    assert_script_run(q{echo 'polkit.addRule(function(action, subject) { polkit.log("action=" + action); polkit.log("subject=" + subject); return polkit.Result.NOT_HANDLED; });' > /etc/polkit-1/rules.d/10-debugging.rules}, 30);
```

Reply:
No need, though, this seems fine.

Reply:
As much as string quoting is ugly, I think I might prefer to keep this more like a reasonable programming language and less esoterically Perl. But this is good to know in general.